addons/im_livechat/models/discuss_channel_member_history.py

```diff
@@ -0,0 +1,34 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields
+
+
+class DiscussChannelMemberHistory(models.Model):
+    _name = "discuss.channel.member.history"
+    _description = "Keep the channel member history"
+
+    channel_id = fields.Many2one("discuss.channel")
+    partner_id = fields.Many2one("res.partner")
+    guest_id = fields.Many2one("mail.guest")
+    member_id = fields.Many2one("discuss.channel.member", ondelete="set null")
+    type = fields.Selection([
+        ("operator", "Operator"),
```

Comment:
We're very likely to rename Operator into Agent, we might want to clarify with AJU if we should use the new term going forward.

Reply:
Shouldn't we make a PR to rename everything at once when we agree on the name?